File: TestApp/app/src/main/java/com/testapp/MainActivity.java
package com.testapp;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.animation.AlphaAnimation;
import android.view.animation.Animation;
import android.widget.Button;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    EditText textView;
    Button button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textView = findViewById(R.id.textView);
        button = findViewById(R.id.button);

        final SharedPreferences preferences = getPreferences(Context.MODE_PRIVATE);
        String oldItem = preferences.getString("oldItem", "Nothing created yet...");

        textView.setText(oldItem);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                SharedPreferences.Editor editor = preferences.edit();
                editor.putString("oldItem", textView.getText().toString());
                editor.apply();

                Animation animation = new AlphaAnimation(1.0f, 0.0f);
                animation.setDuration(1000);
                button.startAnimation(animation);

                Log.d("DEBUG", "Button clicked");
            }
        });


    }
}

File: Kotlin Demo Projects/TweenAnimation/app/src/main/java/com/tweenanimation/MainActivity.kt
package com.tweenanimation

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.animation.AnimationUtils
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val animaFadeIn = AnimationUtils.loadAnimation(this, R.anim.spaceship_anim)
        imageViewShip.startAnimation(animaFadeIn)


    }
}
